price-feeder/oracle/filter.go

```diff
@@ -122,14 +126,15 @@ func FilterCandleDeviations(
 				threshold = deviationThresholds[base]
 			}
 
-			if _, ok := deviations[base]; !ok ||
-				(price.GTE(means[base].Sub(deviations[base].Mul(threshold))) &&
-					price.LTE(means[base].Add(deviations[base].Mul(threshold)))) {
-				if _, ok := filteredCandles[providerName]; !ok {
-					filteredCandles[providerName] = make(map[string][]provider.CandlePrice)
+			if d, ok := deviations[base]; !ok ||
+				(price.GTE(means[base].Sub(d.Mul(threshold))) &&
+					price.LTE(means[base].Add(d.Mul(threshold)))) {
```

Comment:
I know it was not done in this PR, but this is getting confusing, could we move to a separate function?

Reply:
The comparison (if condition)? added: 71dffcf8589590635c5d23ac5d22d3a949cb430b